src/eve-server/market/MarketBotMgr.cpp

```diff
@@ -15,35 +14,89 @@
 #include "market/MarketMgr.h"
 #include "market/MarketProxyService.h"
 
+// ---marketbot update; everything past this point has been completely changed.
+#include "market/MarketDB.h"
+#include "inventory/ItemType.h"
+#include "inventory/ItemFactory.h"
+#include "inventory/InventoryItem.h"
+#include "station/StationDataMgr.h"
+#include "system/SystemManager.h"
+#include "system/SystemEntity.h"
+#include <random>
+#include <cstdint>
+#include <chrono>
 
-static const char* const BOT_CONFIG_FILE = EVEMU_ROOT "/etc/MarketBot.xml";
+extern SystemManager* sSystemMgr;
 
+static constexpr int64 FILETIME_TICKS_PER_DAY = 864000000000;  // 100ns ticks per day; for expelorders to be removed prematurally
 
-MarketBotDataMgr::MarketBotDataMgr()
-{
-    m_initalized = false;
-}
+static const uint32 MARKETBOT_MAX_ITEM_ID = 30000;
+static const std::vector<uint32> VALID_GROUPS = {
+    // Ores & Mining
+    18,                                    // Minerals
+    450, 451, 452, 453, 454, 455, 456,     // Raw ores (part 1)
+    457, 458, 459, 460, 461, 462,
+    465, 466, 467, 468, 469,              // Raw ores (part 2)
+    479,                                  // Scanner Probe
+    482,                                  // Mining Crystals
+    492,                                  // Survey Probe
+    538,                                  // Data Miners
+    548,                                  // Interdiction Probe
+    663,                                  // Mercoxit Mining Crystals
 
-int MarketBotDataMgr::Initialize()
-{
-    m_initalized = true;
+    // Ammo / Charges
+    83,                                   // Projectile Ammo
+    84,                                   // Missiles
+    85,                                   // Hybrid Charges
+    86,                                   // Frequency Crystals
+    87,                                   // Cap Booster Charges
+    88,                                   // Defender Missiles
+    89,                                   // Torpedoes
+    90,                                   // Bombs
+    92,                                   // Mines
+    372, 373, 374, 375, 376, 377,         // Advanced Ammo
+    384, 385, 386, 387, 388, 389,         // Extended Missiles (part 1)
+    390, 391, 392, 393, 394, 395, 396,    // Extended Missiles (part 2)
+    648,                                  // Advanced Rocket
+    653, 654, 655, 656, 657,              // Advanced Missiles
+    772,                                  // Assault Missiles
 
-    /* load current data */
+    // Boosters for Implants
+    303
+};
 
-    sLog.Blue(" MarketBotDataMgr", "Market Bot Data Manager Initialized.");
-    return 1;
+static const char* const BOT_CONFIG_FILE = "/src/utils/config/MarketBot.xml";
```

Comment:
This needs to be updated to be a relative path just like how eve-server.xml is read.

Specifically, I would say don't hardcode this at all and instead define it in EVEServerConfig just like how files.logSettings is defined currently.